File: Frontend/src/types/attachment.ts
export interface Attachment {
  id: number;
  taskItemId: number;
  fileName: string;
  fileUrl: string;
  contentType: string;
  fileSize: number;
  createdAt?: string;
}

export interface CreateAttachmentRequest {
  taskItemId: number;
  fileName: string;
  fileUrl: string;
  contentType: string;
  fileSize: number;
}

File: Backend/CRM.API/CRM.API/Models/Attachment.cs
namespace CRM.API.Models
{
    public class Attachment
    {
        public int Id { get; set; }

        public int TaskItemId { get; set; }

        public string FileName { get; set; } = string.Empty;

        public string FileUrl { get; set; } = string.Empty;

        public string? ContentType { get; set; }

        public long? FileSize { get; set; }

        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

        public TaskItem TaskItem { get; set; } = null!;
    }
}